lldb/source/Host/posix/ProcessLauncherPosixFork.cpp

```diff
@@ -122,8 +123,14 @@ struct ForkLaunchInfo {
         ExitWithError(error_fd, "close");
       break;
     case FileAction::eFileActionDuplicate:
-      if (dup2(action.fd, action.arg) == -1)
-        ExitWithError(error_fd, "dup2");
+      if (action.fd != action.arg) {
```

Comment:
In what circumstances are action.fd and action.arg the same, or not the same?

Reply:
This says "I want to pass `action.fd` from *my* process as `action.arg` to the *new* process". They're the same if the caller specifies them as such, except that we now have special handling to ensure that the O_CLOEXEC flag is cleared in the new process (and only in that process).